src/graph/graph-component.cc

```diff
@@ -140,7 +140,10 @@ namespace hpp {
 
       void GraphComponent::throwIfNotInitialized () const
       {
-        if (!isInit_ || (graph_.lock() && !graph_.lock()->isInit_)) throw std::logic_error ("The graph should have been initialized first.");
+        if (!isInit_){
```

Comment:
I think you need to check the graph isInit_ value. Why did you remove it ?

Reply:
It was necessary. To modifiy the projector of an edge, the edge needs to be initialized. As I modify the projector of previously initialized edges in the initialization of the waypoint edge, the graph is not initialzed yet.

Reply:
The logic behind the isInit_ would be broken by this change because the constraints added to the graph are used in all the other components.

You could add a virtual method `GraphComponent::invalidate` that will set `isInit_ = false`. Then, reimplement it `Graph::invalidate` that calls `invalidate` on each GraphComponent of the graph (except itself, obviously). Then, in all places where you find `isInit_ = false`, call `invalidate` instead.